code/modules/antagonists/cult/blood_magic.dm

```diff
@@ -672,89 +683,127 @@
+	var/mob/living/basic/construct/construct_thing = target
+	var/missing_health = construct_thing.maxHealth - construct_thing.health
+	if(!missing_health)
+		construct_thing.balloon_alert(user, "no damage!")
+		return FALSE
+
+	if(uses <= 0)
+		construct_thing.balloon_alert(user, "out of blood!")
+		return FALSE
+	if(uses > missing_health)
+		construct_thing.adjust_health(-missing_health)
+		construct_thing.visible_message(span_warning("[construct_thing] is fully healed by [user]'s blood magic!"))
+		uses -= missing_health
+	else
+		construct_thing.adjust_health(-uses)
+		construct_thing.visible_message(span_warning("[construct_thing] is partially healed by [user]'s blood magic!"))
+		uses = 0
+	playsound(get_turf(construct_thing), 'sound/magic/staff_healing.ogg', 25)
+	user.Beam(construct_thing, icon_state="sendbeam", time = 1 SECONDS)
+	return TRUE
+
+/**
+	* blood rites cultist healing procedure
+	*
+	**blood healing stuff:
+	**blood_needed = how much blood the target needs to return to safe levels
+	**USES_TO_BLOOD = how many units of blood one "use" of the spell is worth
+	**blood_bank = how many units of blood we can restore
+	*
+	**damage healing stuff:
+	**overall_damage = sum of brute, toxin, fire, and oxygen damage
+	**healing_ratio = percentage of damage healed, between 0 and 1 (0-100%)
+	**healing_cost = how many spell uses we lose due to healing
+*/
+/obj/item/melee/blood_magic/manipulator/proc/heal_cultist(mob/living/carbon/human/human_bloodbag, mob/living/carbon/human/user)
+	if(uses <= 0)
+		human_bloodbag.balloon_alert(user, "out of blood!")
+		return FALSE
+
+	// here we handle blood restoration for those cultists who need it
+	if(human_bloodbag.blood_volume < BLOOD_VOLUME_SAFE)
+		var/blood_needed = BLOOD_VOLUME_SAFE - human_bloodbag.blood_volume
+		var/blood_bank = USES_TO_BLOOD * uses
+		if(blood_bank < blood_needed)
+			human_bloodbag.blood_volume += blood_bank
+			to_chat(user,span_danger("You use the last of your blood rites to restore what blood you could!"))
+			uses = 0
+			return TRUE
+		human_bloodbag.blood_volume = BLOOD_VOLUME_SAFE
+		uses -= round(blood_needed / USES_TO_BLOOD)
+		to_chat(user,span_warning("Your blood rites have restored [human_bloodbag == user ? "your" : "[human_bloodbag.p_their()]"] blood to safe levels!"))
+
+	// and here we heal hurt cultists
+	var/overall_damage = human_bloodbag.getBruteLoss() + human_bloodbag.getFireLoss() + human_bloodbag.getToxLoss() + human_bloodbag.getOxyLoss()
+	if(overall_damage == 0)
+		to_chat(user,span_cult("That cultist doesn't require healing!"))
+		return FALSE
+
+	var/healing_ratio = uses / overall_damage
+	var/healing_cost = round(overall_damage)
+	if(human_bloodbag == user)
+		to_chat(user,span_cult("<b>Your blood healing is far less efficient when used on yourself!</b>"))
+		healing_ratio *= 0.35 // self-healing is 65% less effective
+		healing_cost *= 1.65  // self-healing is 65% more expensive
```

Comment:
Above previously was: 
```
ratio *= 0.35 // Healing is half as effective if you can't perform a full heal
uses -= round(overall_damage) // Healing is 65% more "expensive" even if you can still perform the full heal
```
... which doesn't make sense. I disregarded the top comment and made the bottom one reality in my implementation. Formerly (as I understood it), healing ratio was 35% as effective on yourself and cost twice as much (overall damage was subtracted twice). 

I've maintained the 35% and changed the cost to be 65% more expensive.

Reply:
Upon my review my math isn't right here, marking as draft until I have time to fix.